src/processing/ParameterizedOperationsUtil.java

```diff
@@ -3,195 +3,435 @@
+		}
+
+		// storing data members that remain same: objectId, and timestamp of object creation
+		ObjectId newObjectId = obj.getObjectId();
+		Timestamp newTime = obj.getTimestamp();
+
+		// set (ROTATE) as the operation which is applied on object.
+		IBoardObjectOperation newBoardOp = new RotateOperation(angleOfRotation);
+
+		BoardObject dummyObj = null;
+		try {
+
+			// remove previous object from maps.
+			dummyObj = ClientBoardState.maps.removeObjectFromMaps(obj.getObjectId());
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotationU: old object deleted"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotationU: object deletion failed!"
+			);
+		}
+
+		BoardObject newObj = null;
+		try {
+
+			// create a new object with same objectId, timestamp and other updated values.
+			newObj = CurveBuilder.drawCurve(newPixelSet, newBoardOp, newObjectId, newTime,
+			id, prevPixelIntensities, false);
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotationU: object recreated"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "rotationU: object recreation failed!"
+			);
+		}
+
+		return newObj;
+	}
+
+	/**
+	 * Function for implementing color-change of select-able objects.
+	 */
+	public static BoardObject colorChange(BoardObject obj, UserId id, Intensity intensity) {
+
+		BoardObject newObj = null;
+		try {
+
+			newObj = colorChangeUtil(obj, id, intensity);
+			stackUtil(newObj);
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: operation completed"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: operation failed!"
+			);
+		}
+
+		try {
+
+			// To send all the pixel updates to UI
+			CommunicateChange.provideChanges(newObj.getPrevIntensity(), newObj.getPixels());
+
+			// To send selection updates to UI
+			CommunicateChange.identifierToHandler.get(CommunicateChange.identifierUI).giveSelectedPixels(newObj.getPixels());
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.SUCCESS,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: updates sent to UI"
+			);
+		} catch (Exception e) {
+
+			paraOpLogger.log(
+				ModuleID.PROCESSING, 
+				LogLevel.ERROR,
+				"[#" + Thread.currentThread().getId() + "] "
+				+ "colorChange: UI updates failed!"
+			);
+		}
+
+	// change `boardOp` of old object and return it
+	IBoardObjectOperation newBoardOp = new ColorChangeOperation(intensity);
```

Comment:
372, 373, 374, 376 lines should be indented to the level of other statements in the function (4 spaces in this case)